app/DTO/ArrayableDTO.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace App\DTO;
+
+use App\Exceptions\Internal\LycheeLogicException;
+use Illuminate\Contracts\Support\Arrayable;
+use ReflectionClass;
+use ReflectionProperty;
+use function Safe\sprintf;
+
+/**
+ * In some cases, when a DTO does not need to apply casts on attributes
+ * we can directly make use of a reflection which returns an array containing
+ * all the PUBLIC attributes of the DTO.
+ */
+class ArrayableDTO extends DTO
+{
+	/**
+	 * By default, we return an array containing the PUBLIC attributes of the DTO.
+	 *
+	 * @return array the serialized properties of this object
+	 */
+	public function toArray(): array
+	{
+		$result = [];
+		$cls = new ReflectionClass($this);
+		$props = $cls->getProperties(ReflectionProperty::IS_PUBLIC);
+		foreach ($props as $prop) {
+			$value = $prop->getValue($this);
+			if (is_object($value)) {
+				if ($value instanceof Arrayable) {
+					$value = $value->toArray();
+				} else {
+					throw new LycheeLogicException(sprintf("We don't know how to %s.", get_class($value)));
```

Comment:
This exception message is missing a verb. Moreover, my proposal "we don't know how" was meant sloppily not as a serious exception message.
```suggestion
					throw new LycheeLogicException(sprintf("Unable to convert %s into an array", get_class($value)));
```